components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/ExportUtils.java

```diff
@@ -0,0 +1,1087 @@
+     * @param tenantId     Tenant id of the user
+     * @param provider     Api Provider
+     * @param exportFormat Export format of file
+     * @throws APIImportExportException If an error occurs when writing to file or retrieving certificate metadata
+     */
+    public static void addClientCertificatesToArchive(String archivePath, Identifier identifier,
+                                                      int tenantId, APIProvider provider,
+                                                      ExportFormat exportFormat) throws APIImportExportException {
+        List<ClientCertificateDTO> certificateMetadataDTOs;
+        try {
+            if (identifier instanceof APIProductIdentifier) {
+                certificateMetadataDTOs = provider.searchClientCertificates(tenantId, null,
+                        (APIProductIdentifier) identifier);
+            } else {
+                certificateMetadataDTOs = provider.searchClientCertificates(tenantId, null,
+                        (APIIdentifier) identifier);
+            }
+            if (!certificateMetadataDTOs.isEmpty()) {
+                String clientCertsDirectoryPath = archivePath + File.separator
+                        + ImportExportConstants.CLIENT_CERTIFICATES_DIRECTORY;
+                CommonUtil.createDirectory(clientCertsDirectoryPath);
+
+                JsonArray certificateList = getClientCertificateContentAndMetaData(certificateMetadataDTOs, clientCertsDirectoryPath);
+
+                if (certificateList.size() > 0) {
+                    writeDtoToFile(clientCertsDirectoryPath + ImportExportConstants.CLIENT_CERTIFICATE_FILE,
+                            exportFormat, ImportExportConstants.TYPE_CLIENT_CERTIFICATES, certificateList);
+                }
+            }
+        } catch (IOException e) {
+            String errorMessage = "Error while saving as YAML or JSON";
+            throw new APIImportExportException(errorMessage, e);
+        } catch (APIManagementException e) {
+            String errorMsg = "Error retrieving certificate meta data. tenantId [" + tenantId + "] api ["
+                    + tenantId + "]";
+            throw new APIImportExportException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Get Client Certificate MetaData and Certificate detail and build JSON list.
+     *
+     * @param clientCertificateDTOs client certificates list DTOs
+     * @param certDirectoryPath     directory path to export the certificates
+     * @return list of certificate detail JSON objects
+     */
+    private static JsonArray getClientCertificateContentAndMetaData(List<ClientCertificateDTO> clientCertificateDTOs,
+                                                                    String certDirectoryPath) {
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        JsonArray certificatesList = new JsonArray();
+        clientCertificateDTOs.forEach(metadataDTO -> {
+            ByteArrayInputStream certificate = null;
```

Comment:
Use the stream with `try with resources`

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9406/commits/150d9abd9d4eb25c7f1217fc53f6143f94167542